### ISSUE TITLE
principal.rb:18: warning: deprecated Object#=~ is called on Hash; it always returns nil

### ISSUE THREAD

**Issue by danbryan** [stelligent/cfn_nag#406]
It seems that when I use any Intrinsic function inside a Principal block cfn_nag gives this error. However, if I remove the Intrinsic functions it works just fine.  Cloudformation is happy either way. Any thoughts?

```
S3ArtifactBucketPolicy:
    Type: AWS::S3::BucketPolicy
    Description: Setting Amazon S3 bucket policy for AWS CodePipeline access
    Properties:
      Bucket: !Ref S3Bucket
      PolicyDocument:
        Id: SSEAndSSLPolicy
        Statement:
        - Action:
          - s3:GetBucketVersioning
          Condition:
            Bool:
              aws:SecureTransport: true
          Effect: Allow
          Principal:
            AWS:
              # - !GetAtt CodePipelineTrustRole.Arn
              # - !GetAtt CodeBuildRole.Arn
              - "*"
          Resource:
          - !Sub arn:${AWS::Partition}:s3:::${S3Bucket}
        Version: 2012-10-17
```

**Comment by erickascic** [comment]
@danbryan do you have the error?

**Comment by danbryan** [comment]
[Error - 5:50:06 PM] /home/dan/.gem/ruby/2.7.0/gems/cfn-model-0.4.21/lib/cfn-model/model/principal.rb:18: warning: deprecated Object#=~ is called on Hash; it always returns nil

**Comment by erickascic** [comment]
@danbryan thx for finding this.  the model isn't expecting a reference (Hash) there.  will fix shortly

**Comment by danbryan** [comment]
sounds good, thanks!

**Comment by erickascic** [comment]
this is a "feature" of 2.7 :)

**Comment by erickascic** [comment]
should be fixed in 0.5.26.  thanks!
